Civi/Api4/Action/Address/AddressSaveTrait.php

```diff
@@ -47,14 +47,16 @@ trait AddressSaveTrait {
   /**
    * @inheritDoc
    */
-  protected function writeObjects(&$items) {
-    foreach ($items as &$item) {
+  protected function writeToDao(array $items) {
```

Comment:
I'd like to bike shed this function name a little - I found it really confusing and we are not actually writing to the dao as it implies. We are 'saving' or perhaps 'writeToDatabase'

If 'save' is contaminated? maybe 'saveRecords' ?

Reply:
Ok. Just to frame the issue, what's going on in this PR is that one function is getting split into two smaller functions. The function is `writeObjects()` which previously would:

1) format the params
2) call the appropriate BAO create/add/writeRecords function
3) Reformat and return the results

Now it does 1 and 3, with 2 being split off to this new sub-function. For compatibility reasons I don't think the outer `writeObjects` function should be renamed. So we need to pick a name for the inner sub-function. Something to indicate that it's called from `writeObjects` and is the piece that takes care of the actual writing (or rather, handing it off to the BAO to do the writing). So I named it `writeToDao`.

Reply:
@colemanw what about `writeToDatabase` - alternativelyperhap  `writeUsingDAO` - except it would really be `writeUsingBAOExceptIfThereIsn'tOne` 

Or even just ... `write`

The `writeToDAO` just doesn't seem to be what is actually being done - it's not being written to the DAO - and in some fantasy future world we could replace the DAO layer altogether.....

Reply:
Yea, `write` sounds good.